### DISCUSSION TITLE
Integration Critical CSS.

### DISCUSSION THREAD

**Discussion by OurDevGit** [hayes0724/shopify-packer#82] [Q&A]
Hi, there?
We built the theme using Shopify packer and needed to integrate Critical CSS with the theme.

We tried with `critical-css-webpack-plugin` package but still have some issues.

So if you have any experiences in this, please help us.

- Where I have to add webpack configration?  (dev.config.js ? or packer.config.js? or prod.config.js? )
- Which options we should put and what is it ? (src and target? )
- If you have anyother experiences, please let me know what is the webpack critical package.

Hope hear from you soon.

Thanks.

**Comment by mrpacman101** [comment]
If anyone has a working solution or suggestions - we would highly appreciate it, how this would be implemented with packer.

**Comment by hayes0724** [comment]
@OurDevGit @mrpacman101 I'm looking into adding recipes and plugins for easy modification. For now you can make any edits by overriding the dev and prod webpack configs.

I don't have much experience with `critical-css-webpack-plugin` but the files you would need to edit are:

 `dev.config.js`
[Dev config](https://github.com/hayes0724/shopify-packer/blob/master/src/webpack/config/dev.config.js)
 
 `prod.config.js`
[Production config](https://github.com/hayes0724/shopify-packer/blob/master/src/webpack/config/prod.config.js)

Uses [webpack-merge](https://www.npmjs.com/package/webpack-merge)
